cmd/swarmctl/service/flagparser/port.go

```diff
@@ -30,6 +30,9 @@ func parsePorts(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
 			Protocol:      protocol,
 			TargetPort:    port,
 			PublishedPort: swarmPort,
+			// In swarmctl all ports are by default
+			// PublishModeHost
+			PublishMode: api.PublishModeHost,
```

Comment:
This only affects `swarmctl`. As it is swarmctl/swarmd cannot support ingress ports since there is no necessary engine-api which is exposed to achieve this neither will it be any time soon. With the introduction of Host PublishMode the only Mode that can be supported is Host and so we always only create only one kind of ports via `swarmctl`.